tests/snuba/api/endpoints/test_organization_events_v2.py

```diff
@@ -3779,6 +3779,50 @@ def test_team_key_transactions_query(self):
         assert data[0]["team_key_transaction"] == 0
         assert data[0]["transaction"] == "/blah_transaction/"
 
+    def test_too_many_team_key_transactions(self):
+        teams = [
+            self.create_team(organization=self.organization, name=f"Team {i}") for i in range(10)
+        ]
+
+        for team in teams:
+            self.create_team_membership(team, user=self.user)
+            self.project.add_team(team)
+
+        TeamKeyTransaction.objects.bulk_create(
+            [
+                TeamKeyTransaction(
+                    team=team,
+                    organization=self.organization,
+                    transaction=f"transaction-{team.id}-{i}",
+                    project=self.project,
+                )
+                for team in teams
+                for i in range(MAX_TEAM_KEY_TRANSACTIONS)
+            ]
+        )
```

Comment:
Could we test this in a way where we don't have to create 500 entries in the db?